core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -30,96 +54,209 @@
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, String... values) {
-        this(name, (Object[]) values);
+    public TermsQueryBuilder(String fieldName, String... values) {
+        this(fieldName, values != null ? Arrays.asList(values) : null);
     }
-
+    
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, int... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, int... values) {
+        this(fieldName, values != null ? Ints.asList(values) : null);
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, long... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, long... values) {
+        this(fieldName, Longs.asList(values));
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, float... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, float... values) {
+        this(fieldName, Floats.asList(values));
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, double... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, double... values) {
+        this(fieldName, Doubles.asList(values));
     }
 
     /**
      * A filter for a field based on several terms matching on any of them.
      *
-     * @param name   The field name
+     * @param fieldName The field name
      * @param values The terms
      */
-    public TermsQueryBuilder(String name, Object... values) {
-        this.name = name;
-        this.values = values;
+    public TermsQueryBuilder(String fieldName, Object... values) {
+        this(fieldName, values != null ? Arrays.asList(values) : null);
+    }
+
+    /**
+     * Constructor used for terms query lookup.
+     *
+     * @param fieldName The field name
+     */
+    public TermsQueryBuilder(String fieldName) {
+        this.fieldName = fieldName;
```

Comment:
Maybe call this(fieldname, null) here just so we have as few entry points for creating this class as possible?